src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -0,0 +1,217 @@
+            this.bitsPerEntry = bitsPerEntryIn;
+            this.maxEntryValue = (1L << bitsPerEntryIn) - 1L;
+
+            if (longArrayIn != null)
+            {
+                this.longArray = longArrayIn;
+            }
+            else
+            {
+                this.longArray = new long[(int) (roundUp((long) arraySizeIn * (long) bitsPerEntryIn, 64L) / 64L)];
+            }
+        }
+
+        public void setAt(long index, int value)
+        {
+            Validate.inclusiveBetween(0L, this.arraySize - 1L, (long) index);
+            Validate.inclusiveBetween(0L, this.maxEntryValue, (long) value);
+            long startOffset = index * (long) this.bitsPerEntry;
+            int startArrIndex = (int) (startOffset >> 6); // startOffset / 64
+            int endArrIndex = (int) (((index + 1L) * (long) this.bitsPerEntry - 1L) >> 6);
+            int startBitOffset = (int) (startOffset & 0x3F); // startOffset % 64
+            this.longArray[startArrIndex] = this.longArray[startArrIndex] & ~(this.maxEntryValue << startBitOffset) | ((long) value & this.maxEntryValue) << startBitOffset;
+
+            if (startArrIndex != endArrIndex)
+            {
+                int endOffset = 64 - startBitOffset;
+                int j1 = this.bitsPerEntry - endOffset;
+                this.longArray[endArrIndex] = this.longArray[endArrIndex] >>> j1 << j1 | ((long) value & this.maxEntryValue) >> endOffset;
+            }
+        }
+
+        public int getAt(long index)
+        {
+            Validate.inclusiveBetween(0L, this.arraySize - 1L, (long) index);
+            long startOffset = index * (long) this.bitsPerEntry;
+            int startArrIndex = (int) (startOffset >> 6); // startOffset / 64
+            int endArrIndex = (int) (((index + 1L) * (long) this.bitsPerEntry - 1L) >> 6);
+            int startBitOffset = (int) (startOffset & 0x3F); // startOffset % 64
+
+            if (startArrIndex == endArrIndex)
+            {
+                return (int) (this.longArray[startArrIndex] >>> startBitOffset & this.maxEntryValue);
+            }
+            else
+            {
+                int endOffset = 64 - startBitOffset;
+                return (int) ((this.longArray[startArrIndex] >>> startBitOffset | this.longArray[endArrIndex] << endOffset) & this.maxEntryValue);
+            }
+        }
+
+
+        public long size()
+        {
+            return this.arraySize;
+        }
+
+        public static long roundUp(long number, long interval)
+        {
+            if (interval == 0)
+            {
+                return 0;
+            }
+            else if (number == 0)
+            {
+                return interval;
+            }
+            else
+            {
+                if (number < 0)
+                {
+                    interval *= -1;
+                }
+
+                long i = number % interval;
+                return i == 0 ? number : number + interval - i;
+            }
+        }
```

Comment:
```suggestion
        public static long roundUp(double number, long interval)
        {
            if (number < 0){
                interval *= -1;
            }
            return interval * (long) Math.ceil(number/interval);
        }
```

I believe that this has exactly the same functionality, if you have any reason for wanting to use this instead please do, however I think this is nicer,

Reply:
No, it will throw an exception instead of returning 0 if interval is 0 and if number is 0 it returns 0 instead of number.

Reply:
I don't believe that interval is ever 0, the only occurrence I could found was that it was 64.

Reply:
I can test it. The nested class there is straight out of litematica, so I just left the code as-is.